src/obfuscator.cpp

```diff
@@ -0,0 +1,64 @@
+// Unless explicitly stated otherwise all files in this repository are
+// dual-licensed under the Apache-2.0 License or BSD-3-Clause License.
+//
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2021 Datadog, Inc.
+
+#include <obfuscator.hpp>
+#include <utils.h>
+
+namespace ddwaf
+{
+
+namespace
+{
+
+bool match(re2::RE2 &regex, std::string_view value)
+{
+    size_t length = findStringCutoff(value.data(), value.size());
+    const re2::StringPiece key_ref(value.data(), length);
+
+    return regex.Match(key_ref, 0, length, re2::RE2::UNANCHORED, nullptr, 0);
+}
+
+}
+
+obfuscator::obfuscator(std::string_view key_regex_str,
```

Comment:
What happens if the regexp cannot be compiled?
The lib should know so we can log an error.
I even wonder if we should start appsec in such as case.

Reply:
Literally nothing happens at the moment because RE2 fails silently. We can certainly log an error.

Reply:
I have now added an error log if the regex fails to compile.

Reply:
As discussed, I've changed it so that it'll fallback to the default key regex if it fails to compile the user-provided one and if that one fails to compile it'll fail to initialise the WAF.